### PR TITLE
Unify ThreadX and SMP for ARMv8-A.

### PR DESCRIPTION
**Pull request by TiejunMS**
## PR checklist

- [x] Updated function header with a short description and version number
- [ ] Added test case for bug fix or new feature
- [ ] Validated on real hardware

### CODE REVIEW

**Comment by hodcarrier** [comment]
hello, ARMv8-A is hard to type... other projects refers it as armv8-a, gcc's compile option is -march=armv8-a, so why make a new name...

**Comment by TiejunMS** [comment]
> hello, ARMv8-A is hard to type... other projects refers it as armv8-a, gcc's compile option is -march=armv8-a, so why make a new name... when pass the armv8-a from a environment variable, you have to do one more step: translate armv8-a to ARMv8-A eg. if [ THREADX_ARCH == ARMv8-A ] CFLAGS += -march=armv8-a end of if

Thanks for the suggestion! But ports_arch is a folder for internal development only. Users should use files from ports, ports_module and ports_smp.

File: ports/cortex_a35/gnu/example_build/tx/.cproject [hunk 1]
```diff
@@ -39,7 +39,7 @@
                             							
                             <tool id="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.var.gcc-8.3.0-aarch64-elf.1755052156" name="GCC C Compiler 8.3.0 [aarch64-elf]" superClass="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.var.gcc-8.3.0-aarch64-elf">
                                 								
-                                <option id="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.option.cpu.281590127" name="CPU (-mcpu)" superClass="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.option.cpu" useByScannerDiscovery="false" value="cortex-a35" valueType="string"/>
+                                <option id="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.option.cpu.281590127" name="CPU (-mcpu)" superClass="com.arm.eclipse.cdt.managedbuild.ds5.gcc.baremetal.tool.c.compiler.option.cpu" useByScannerDiscovery="false" value="" valueType="string"/>
```
Comment (bo-ms) [inline ports/cortex_a35/gnu/example_build/tx/.cproject:42]: Could you explain why the value is set as null? thanks

**Review by bo-ms — APPROVED** [review]

File: ports/cortex_a35/gnu/example_build/tx/.cproject (hunk 1 above)
Comment (TiejunMS) [reply ports/cortex_a35/gnu/example_build/tx/.cproject:42]: Use -march to support generic CPU types and simplifies the maintenance.

**Pull request merged**